Core/Aggregates/IAggregate.cs

```diff
@@ -2,14 +2,13 @@
 
 namespace Core.Aggregates;
 
-public interface IAggregate: IAggregate<Guid>
+public interface IAggregate: IProjection
 {
-}
-
-public interface IAggregate<out T>: IProjection
-{
-    T Id { get; }
     int Version { get; }
```

Comment:
a repository knows the version it saw when it loaded a batch of events (and then it filters and upconverted etc them before passing them on)
it also knows the version after the update cycle has completed